int64 msg_int64
bool msg_bool
string msg_string